core/src/main/java/org/apache/accumulo/core/security/Authorizations.java

```diff
@@ -112,8 +124,13 @@ private void checkAuths() {
    */
   public Authorizations(Collection<byte[]> authorizations) {
     checkArgument(authorizations != null, "authorizations is null");
-    for (byte[] auth : authorizations)
+    Pair<Set<ByteSequence>,List<byte[]>> collections =
+        createInternalCollections(authorizations.size());
+    this.auths = collections.getFirst();
+    this.authsList = collections.getSecond();
```

Comment:
We could avoid creating the Pair object by replacing `createInternalCollections` with two private static methods `createInternalSet` and `createInternalList`.

```suggestion
    this.auths = createInternalSet(authorizations.size());
    this.authsList = createInternalList(authorizations.size());
```

Reply:
I went ahead and broke the single method into two, removing the need for Pair. I was using Pair as a struct, trying to minimize the methods called, but I think it may not matter in the long run. Certainly, not allocating a temporary object is better here.